### PR TITLE
Gas limit

### PR DESCRIPTION
**Pull request by sesamegu**

### CODE REVIEW

File: cmd/gtas/cli/gtas.go [hunk 1]
```diff
@@ -308,6 +308,20 @@ func (gtas *Gtas) fullInit() error {
 	common.GlobalConf.SetString(Section, "miner", gtas.account.Address)
 	fmt.Println("Your Miner Address:", gtas.account.Address)
 
+	//set the time for proposer package
+	timeForPackage := common.GlobalConf.GetInt(Section, "time_for_package", 2000)
+	if timeForPackage > 100 && timeForPackage < 2000 {
+		core.ProposerPackageTime = time.Duration(timeForPackage) * time.Microsecond
```
Comment (darren0718) [inline cmd/gtas/cli/gtas.go]: should be time.Duration(timeForPackage) * time.Millisecond!

**Review by darren0718 — CHANGES_REQUESTED** [review]

File: core/blockchain_action.go [hunk 2]
```diff
@@ -156,11 +156,6 @@ func (chain *FullBlockChain) verifyTxs(bh *types.BlockHeader, txs []*types.Trans
 		return -1
 	}
 
-	if !chain.validateTxRoot(bh.TxTree, txs) {
```
Comment (bird512) [inline core/blockchain_action.go:159]: This check should not be removed. need to validate the transactions root hash in a block

File: cmd/gtas/cli/gtas.go (hunk 1 above)
Comment (sesamegu) [reply cmd/gtas/cli/gtas.go]: ok

File: core/blockchain_action.go (hunk 2 above)
Comment (sesamegu) [reply core/blockchain_action.go:159]: ok

File: core/blockchain_action.go [hunk 3]
```diff
@@ -441,10 +441,14 @@ func (chain *FullBlockChain) executeTransaction(block *types.Block) (bool, *exec
 
 	statehash, evitTxs, transactions, receipts, gasFee, err := chain.executor.Execute(state, block.Header, block.Transactions, false, nil)
 	txTree := calcTxTree(transactions)
-	if txTree != block.Header.TxTree{
+	if txTree != block.Header.TxTree {
 		Logger.Errorf("Fail to verify txTree, hash1:%s hash2:%s", txTree, block.Header.TxTree)
 		return false, nil
 	}
+	if len(transactions) != len(block.Transactions) {
```
Comment (darren0718) [inline core/blockchain_action.go:448]: no need to check this, because the txTree check already work

**Review by darren0718 — CHANGES_REQUESTED** [review]

File: core/blockchain_action.go (hunk 3 above)
Comment (sesamegu) [reply core/blockchain_action.go:448]: explained

**Review by darren0718 — APPROVED** [review]

**Pull request merged**